### PR TITLE
fixed for privacy violations

### PR DESCRIPTION
**Pull request by OhWellNevermind**
### Description

A clear and concise summary of the change and which issue (if any) it fixes. Should also include relevant motivation and context.

Resolved or fixed issue: 

### Affirmation

- [ ] My code follows the [CONTRIBUTING.md](https://github.com/juice-shop/juice-shop/blob/master/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: routes/2fa.ts [hunk 1]
```diff
@@ -62,33 +72,31 @@ async function verify (req: Request, res: Response) {
...
+    const { data: user } = data;
 
-    if (user.totpSecret === '') {
-      const secret = otplib.authenticator.generateSecret()
+    if (user.totpSecret === "") {
+      const secret = otplib.authenticator.generateSecret();
 
       res.json({
         setup: false,
-        secret,
```
Comment (OhWellNevermind) [inline routes/2fa.ts:78]: Removed secret and email

File: routes/authenticatedUsers.ts [hunk 2]
```diff
@@ -23,8 +23,6 @@ async function retrieveUserList (req: Request, res: Response, next: NextFunction
 
         return {
           ...user.dataValues,
-          password: user.password?.replace(/./g, '*'),
```
Comment (OhWellNevermind) [inline routes/authenticatedUsers.ts:26]: removed private user data from response

File: frontend/src/app/change-password/change-password.component.ts [hunk 3]
```diff
@@ -3,83 +3,126 @@
...
-      this.error = error
-      this.confirmation = undefined
-      this.resetPasswords()
-    })
+      .subscribe(
+        (response: any) => {
+          this.error = undefined;
+          this.translate.get("PASSWORD_SUCCESSFULLY_CHANGED").subscribe(
+            (passwordSuccessfullyChanged) => {
+              this.confirmation = `Password successfully changed!`;
```
Comment (OhWellNevermind) [inline frontend/src/app/change-password/change-password.component.ts:84]: Changed confirmation message

File: frontend/src/app/change-password/change-password.component.ts [hunk 4]
```diff
@@ -3,83 +3,126 @@
...
+            },
+            (translationId) => {
+              this.confirmation = `Password successfully changed.`;
```
Comment (OhWellNevermind) [inline frontend/src/app/change-password/change-password.component.ts:87]: Here too

File: frontend/src/app/change-password/change-password.component.ts [hunk 5]
```diff
@@ -3,83 +3,126 @@
...
+            }
+          );
+          this.resetForm();
+        },
+        (error) => {
+          this.error = `Failed to change password. Please try again.`;
```
Comment (OhWellNevermind) [inline frontend/src/app/change-password/change-password.component.ts:93]: Changed error message

**Review by IvanKoshan2004 — APPROVED** [review]

**Pull request merged**